src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_FACTOR = 1.5;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForIndex(index);
+        ensureCapacity();
+        for (int i = size; i > index; i--) {
+            elementData[i] = elementData[i - 1];
+        }
+        elementData[index] = value;
+        size++;
 
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list == null || list.isEmpty()) {
+            throw new IllegalArgumentException("List is null! Please make a list with objects.");
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T oldValue = elementData[index];
+        System.arraycopy(elementData, index + 1, elementData, index, size - index - 1);
+        elementData[--size] = null;
+        return oldValue;
+
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        if (element == null) {
+            size--;
+            return null;
+        }
+
+        for (int i = 0; i < size; i++) {
+            if (element.equals(elementData[i])) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("Element not found:  " + element);
```

Comment:
The exception message 'Element not found:  ' + element is not informative enough. It should also include information about the attempted operation, like 'Failed to remove element because it was not found: ' + element.